File: src/java-sample/com/xtaudio/xt/Sample.java
package com.xtaudio.xt;

public class Sample {

    public static void main(String[] args) throws Exception {

        System.out.println("PrintSimple:");
        PrintSimple.main(args);
        System.out.println("PrintDetailed:");
        PrintDetailed.main(args);
        System.out.println("CaptureSimple:");
        CaptureSimple.main(args);
        System.out.println("RenderSimple:");
        RenderSimple.main(args);
        System.out.println("CaptureAdvanced:");
        CaptureAdvanced.main(args);
        System.out.println("RenderAdvanced:");
        RenderAdvanced.main(args);
        System.out.println("FullDuplex:");
        FullDuplex.main(args);
        System.out.println("Aggregate:");
        Aggregate.main(args);
    }
}

File: src/cpp/sample/Sample.cpp
#include <xt/XtAudio.hpp>
#include <string>
#include <cstdlib>
#include <iostream>

extern int AggregateMain();
extern int FullDuplexMain();
extern int PrintSimpleMain();
extern int RenderSimpleMain();
extern int PrintDetailedMain();
extern int CaptureSimpleMain();
extern int RenderAdvancedMain();
extern int CaptureAdvancedMain();

static char const* 
Names[] =
{
  "PrintSimple", "PrintDetailed", "CaptureSimple", "RenderSimple",
  "CaptureAdvanced", "RenderAdvanced", "FullDuplex", "Aggregate"
};

static int(*Samples[])() = 
{
  PrintSimpleMain, PrintDetailedMain, CaptureSimpleMain, RenderSimpleMain,
  CaptureAdvancedMain, RenderAdvancedMain, FullDuplexMain, AggregateMain, 
};

static void 
RunSample(int32_t index)
{
  std::cout << Names[index] << ":\n";
  Samples[index]();
}

int 
main(int argc, char** argv) 
{
  int32_t index = argc == 2? std::stoi(std::string(argv[1])): -1;
  try
  {
    if (index >= 0) RunSample(index);
    else for (int32_t i = 0; i < sizeof(Samples)/sizeof(Samples[0]); i++) RunSample(i);
    return EXIT_SUCCESS;
  } catch (Xt::Exception const& e)
  { 
    std::cout << Xt::Audio::GetErrorInfo(e.GetError()) << "\n"; 
    return EXIT_FAILURE;
  }
}